### PR TITLE
Serialization for FeatureGeneratorStage 

### PR DESCRIPTION
**Pull request by wsuchy**
Only saving to JSON

### CODE REVIEW

File: core/src/main/scala/com/salesforce/op/OpWorkflowModelReader.scala [hunk 1]
```diff
@@ -138,6 +138,8 @@ class OpWorkflowModelReader(val workflowOpt: Option[OpWorkflow]) extends MLReade
         val originalStage = workflow.stages.find(_.uid == stageUid)
         originalStage match {
           case Some(os) => new OpPipelineStageReader(os).loadFromJson(j, path = path).asInstanceOf[OPStage]
+          case None if stageUid.startsWith("FeatureGeneratorStage_") =>
```
Comment (leahmcguire) [inline core/src/main/scala/com/salesforce/op/OpWorkflowModelReader.scala]: classOf[FeatureGeneratorStage].getSimpleName instead of string

File: features/src/main/scala/com/salesforce/op/features/FeatureBuilder.scala [hunk 2]
```diff
@@ -270,33 +374,14 @@ final class FeatureBuilderWithExtract[I, O <: FeatureType]
...
-  /**
-   * Feature aggregation function with zero value of [[FeatureTypeDefaults.default[O].value]]
-   * @param fn aggregation function
-   */
-  def aggregate(fn: (O#Value, O#Value) => O#Value): this.type = {
-    val zero = FeatureTypeDefaults.default[O](tto).value
-    aggregator = new CustomMonoidAggregator[O](associativeFn = fn, zero = zero)(tto)
-    this
-  }
-
```
Comment (leahmcguire) [inline features/src/main/scala/com/salesforce/op/features/FeatureBuilder.scala]: why are these deleted? they have an important function

File: features/src/test/scala/com/salesforce/op/features/FeatureBuilderTest.scala [hunk 3]
```diff
@@ -181,6 +181,9 @@ class FeatureBuilderTest extends FlatSpec with TestSparkContext {
 
 }
 
+object TestCustomMonoidAggregator extends CustomMonoidAggregator[Real](zero = Real.empty.v,
```
Comment (leahmcguire) [inline features/src/test/scala/com/salesforce/op/features/FeatureBuilderTest.scala]: so people need to instantiate the class for the aggregation function?

File: readers/src/main/scala/com/salesforce/op/test/PassengerFeaturesTest.scala [hunk 4]
```diff
@@ -34,36 +34,70 @@ import com.salesforce.op.features.types._
...
   val rawFeatures: Array[OPFeature] = Array(
     survived, age, gender, height, weight, description, boarded, stringMap, numericMap, booleanMap
   )
 
 }
+
+object TestMonoidAggregator
+  extends CustomMonoidAggregator[Real](None, (l, r) => (l -> r).map(breeze.linalg.max(_, _))) with Serializable
+
+object PassengerFeaturesTestLambdas {
```
Comment (leahmcguire) [inline readers/src/main/scala/com/salesforce/op/test/PassengerFeaturesTest.scala]: do you need the functions defined this way or is it a style thing?

Comment (tovbinm) [reply readers/src/main/scala/com/salesforce/op/test/PassengerFeaturesTest.scala]: that the new thing

**Comment by tovbinm** [comment]
@leahmcguire it's WIP, not for review yet ;)

**Comment by tovbinm** [comment]
Superseeded by #309
